Cards/Effects/EffectBase.gd

```diff
@@ -3,14 +3,30 @@ class_name EffectBase extends Resource
 
 @warning_ignore("unused_parameter")
 func apply_effect(caster: Entity, target: Entity, value: int) -> void:
-    pass
+	pass
 
 # the goal of this function is to provide a base targeting
 # here it's not really useful for what is does, but the goal is to return differently
 # in children class. For example an AoE where you return the entity and the entities on its side
 func build_target_list(target: Entity) -> Array[Entity]:
-    return [target]
+	return [target]
 
 # To be called by child classes through apply_effect. But since there might be multiple effects that need it, it's available in the parent class
 func _apply_status(status: StatusBase, caster: Entity, target: Entity) -> void:
-    target.get_status_component().add_status(status, caster)
\ No newline at end of file
+	target.get_status_component().add_status(status, caster)
+
+static func calculate_value_modified_by_stats(enum_name: int, caster: Entity, target: Entity, value: int) -> int:
```

Comment:
can you have an enum as type ?

Reply:
I believe so. I tried this just now on your branch: 

![image](https://github.com/Saplings-Projects/1M_sub/assets/5923102/eadd2a50-57de-454b-b0a0-7e5964fe7009)

![image](https://github.com/Saplings-Projects/1M_sub/assets/5923102/2196215f-c9f2-47b6-a2a3-d934e90f31e4)

The compiler wasn't screaming at me when running the tests, although I'm not sure if this would translate when running the tests.

Reply:
alright, will try that then